src/rez/package_order.py

```diff
@@ -657,31 +678,32 @@ def refresh(self) -> None:
                     continue
                 self.by_package[package] = orderer
 
-    def append(self, *args, **kwargs):
-        self.dirty = True
-        return super().append(*args, **kwargs)
+    if not TYPE_CHECKING:
+        def append(self, *args, **kwargs):
```

Comment:
Since this class inherits from `list` it's easier to rely on the type hints coming from that base class than to redefine them here, so we hide them by placing them behind `not TYPE_CHECKING`.   In reality, the runtime value of `TYPE_CHECKING` is always `False`.